bite-size-notes/week-6-b-free-final-time-PMP.md

```diff
@@ -0,0 +1,231 @@
+
+The purpose behind using the calculus of variations for free final time is to find a extrema function that minimizes/maximizes a functional: analogous to finding the critical points by setting the derivative of a function equal to zero.
+
+Our optimization cost is then represented by a cost functional $J[x(t),u(t),t]$ rather than a cost function $J\big(x(t),u(t),t\big)$. More precisely,
+
+$$\underbrace{J[z(t),u(t),t]}_{\text{cost functional}} = \int_0^T \underbrace{L\big(z(t), u(t), t\big)}_{\text{cost-to-go}} \,dt +  \underbrace{\Psi(T)}_{\text{terminal cost}}$$
+
+where $L$ is the Lagrangian, $\Psi$ is a terminal cost function, and $T$ is the terminal time.
+
+#### Sufficient conditions for optimality
+
+If control is not an optimization function, the Euler-Lagrange equation from calculus of variations is a necessary condition for optimality of a functional.
+
+However, if control is also being optimized, Pontryagin's Minimum Principle gives necessary conditions for minimization of a functional. If transversality conditions are included and the Hamiltonian w.r.t. the state variables is strictly convex, that set of conditions is also sufficient. [1]
+
+### Pontryagin's Minimum Principle (PMP)
+
+Let the cost functional be
+$$\underbrace{J[z(t),u(t),t]}_{\text{cost functional}} = \int_0^T \underbrace{L\big(z(t), u(t), t\big)}_{\text{cost-to-go}} \,dt +  \underbrace{\Psi(T)}_{\text{terminal cost}}$$
+as defined above.
+
+The constraints from the system dynamics are added to the Lagrangian $L$ with time-varying Lagrange multipliers $\lambda(t)$  whose elements are called the co-states of the system.
+
+We can then construct the Hamiltonian $H$:
+$$
+\begin{equation*}
+H\big(z(t), u(t), \lambda(t), t\big) = \lambda(t)^\top \cdot \dot{z} + L\big(z(t), u(t)\big).
+\end{equation*}
+$$
+
+Pontryagin's minimum principle states that the optimal parameters must minimize the Hamiltonian $H$ for all time $t \in [0, T]$ and all feasible controls $u$ so that
+$$
+\begin{equation}
+H(z^*(t), u^*(t), \lambda^*(t), t) \leq H(z(t), u(t), \lambda(t), t).
+\end{equation}
+$$
+
+$$
+\begin{equation}
+\begin{aligned}
+-\dot{\lambda}(t)
+&= \nabla_z H(z^*(t), u^*(t), \lambda^*(t), t) \\ 
+&= \lambda(t)^\top \cdot \nabla_z \dot{z}(z^*(t), u^*(t)) + \nabla_z L(z^*(t), u^*(t)) 
+\end{aligned}
+\end{equation}
+$$
+
+$$
+\begin{equation}
+\lambda(T)^\top = \nabla_z \Psi(z(T))
+\end{equation}
+$$
+
+The above equations are necessary for optimality with fixed final time, this last condition is for free final time.
+$$
+\begin{equation}
+\nabla_T \Psi(z(T)) + H(T) = 0
+\end{equation}
+$$
+
+### Transversality Conditions
+
+For free final time, the transversality conditions state that at the optimal terminal time $T$, the Hamiltonian $H$ should satisfy
```

Comment:
N.B.: the following conditions arise as a consequence of how you choose the terminal cost $\Psi$, thus are a specialization of the previous conditions; could you clarify this aspect?

Reply:
You are correct, I miswrote that part. The source I read gave an example that had no terminal cost and unconstrained final state along with free final time, thus that condition equaled zero.

In general, it's not so, thus only the Hamiltonian being zero at final time is the optimality condition for the free final time extension.